%
%
% UCSD Doctoral Dissertation Template
% -----------------------------------
% http://ucsd-thesis.googlecode.com
%
%


%% REQUIRED FIELDS -- Replace with the values appropriate to you

% No symbols, formulas, superscripts, or Greek letters are allowed
% in your title.
\title{ALE Analytics: A Software Pipeline and Web Platform for the Analysis of Microbial Genomic Data from Adaptive Laboratory Evolution Experiments}

\author{Patrick Phaneuf}
\degreeyear{\the\year}
\campus{San Diego}  % used as: University of California, \campus


% Master's Degree theses will NOT be formatted properly with this file.
\degreetitle{\thedegree}

\field{Computer Science and Engineering}
% \specialization{Anthropogeny}  % If you have a specialization, add it here

\chair{Professor Bernhard Palsson}
% Uncomment the next line iff you have a Co-Chair
% \cochair{Professor Cochair Semimaster}
%
% Or, uncomment the next line iff you have two equal Co-Chairs.
%\cochairs{Professor Chair Masterish}{Professor Chair Masterish}

%  The rest of the committee members  must be alphabetized by last name.
\othermembers{
Professor Vineet Bafna\\
Professor Pavel Pevzner\\
}
 \ifnum\pdfstrcmp{\@degree}{masters}=0 %
    \numberofmembers{3} % |chair| + |cochair| + |othermembers|
\else
    \numberofmembers{3} % |chair| + |cochair| + |othermembers|
\fi

%% START THE FRONTMATTER
%
\begin{frontmatter}

%% TITLE PAGES
%
%  This command generates the title, copyright, and signature pages.
%
\makefrontmatter

\begin{dedication} 
I dedicate this work to my family.
\end{dedication}


%% SETUP THE TABLE OF CONTENTS
%
\tableofcontents
\listoffigures  % Comment if you don't have any figures
\listoftables   % Comment if you don't have any tables

\begin{acknowledgements}
I came to University of California, San Diego, pursuing a dream to apply computer science and engineering towards better understanding and leveraging the fundamentals of biology. I greatly appreciate the opportunities that the university, the department of computer science and engineering and their faculty have offered to enrich this endeavor.

I would like to thank Professor Bernhard Palsson for his support as adviser and the chair of my thesis committee. His ideas, optimism and direction have continuously inspired our broader ambitions for this work. With the Systems Biology Research Group, he has cultivated a group of kind, enthusiastic and driven researchers that continue to impress and innovate.

I would like to thank Professor Vineet Bafna and Professor Pavel Pevzner for their time as committee members of this thesis and for their work as professors to the many courses that have fueled my passions in bioinformatics and systems biology.

I would like to thank Dr. Adam Feist for his mentorship and support. Along with his enormous contributions to this work, he has done everything in his power to enable my success at the University of California, San Diego.

I would like to thank Dennis Gosting; his hard work, initiative and keen insights have greatly enriched the feature set and usability of the products of this thesis.

I would like to thank Dr. Adam Feist, Dr. Ryan LaCroix, Troy Sandberg, Gabriela Guzman, Joon Ho Park, Colton Llyod, Douglas McCloskey and the many others who have contributed their experiment data to enable the work of this thesis.

I would like to thank Dr. Ali Ebrahim, Dr. Ryan LaCroix, Kaiwen Zhang and Dylan Skola; their work lay the foundation for the products of this thesis.

I would like to thank Nadyne Nawar for all of her effort as my master's program adviser in helping me build a personalized academic curriculum that was aligned with my interests and passions.

This work was supported by a grant from the Novo Nordisk Foundation Center for Biosustainability (NNF16CC0021858).

\end{acknowledgements}

%% ABSTRACT
%
%  Doctoral dissertation / thesis abstracts should not exceed 350 words.
%   The abstract may continue to a second page if necessary. Masters thesis abstract should not exceed 250 words according to OGS.
%
\begin{abstract}
%Adaptive Laboratory Evolution (ALE) is a tool for studying biological molecular mechanisms and evolutionary dynamics through coupling with whole genome sequencing. Academic and industrial labs involved in the study of microbial evolution and metabolic engineering use ALE methodologies in their research for exploring adaptive mutations \cite{ASM:/content/journal/microbe/10.1128/microbe.6.69.1}. These labs build custom post-processing and computational pipelines for the analysis of the ALE genomic output, though ALE methodologies often include the same fundamental steps to interrogate the evolutionary trajectory of the organisms under study. This thesis will detail the strategies and computational tools used in the ALE project pipeline executed by UCSD's Systems Biology Research Group (SBRG) and describe a system for compiling and analyzing ALE results. This thesis can contribute to the domain of ALE methodologies by detailing an approach to the computational analysis of ALE experiments, will propose a method for hypothesizing key mutations and will explore global trends for all ALE experiment data accumulated. These computational tools and protocols to identifying key mutations and global trends will be leveraged, implemented and evaluated in our ALE Analytics software pipeline and web platform. We therefore hypothesize that we can leverage existing ALE foundational technologies and methodologies to develop a bioinformatics system that will accurately predict the key mutations of ALE experiments and provide context on global mutational trends and correlated experimental conditions.
Adaptive Laboratory Evolution (ALE) methodologies are used for studying microbial adaptive mutations that optimize host metabolism. The Systems Biology Research Group (SBRG) at the University of  California, San Diego, has implemented high-throughput ALE experiment automation that enables the group to expand their experimental evolutions to scales previously infeasible with manual workflows. The data generated by the high-throughput automation now requires a post-processing, content management and analysis framework that can operate on the same scale. We developed a software system which solves the SBRG's specific ALE big data to knowledge challenges. The software system is comprised of a post-processing protocol for quality control, a software framework and database for data consolidation and a web platform named ALE Analytics for report generation and automated key mutation analysis. The automated key mutation analysis is evaluated against published ALE experiment key mutation results from the SBRG and maintains an average recall of 89.6\% and an average precision of 71.2\%. The consolidation of all ALE experiments into a unified resource has enabled the development of web applications that compare key mutations across multiple experiments. These features find the genomic regions \textit{rph}, \textit{hns/tdk}, \textit{rpoB}, \textit{rpoC} and \textit{pykF} mutated in more than one ALE experiment published by the SBRG. We reason that leveraging this software system relieves the bottleneck in ALE experiment analysis and generates new data mining opportunities for research in understanding system-level mechanisms that govern adaptive evolution.
\end{abstract}


\end{frontmatter}
